Source/Core/Core/AchievementManager.cpp

```diff
@@ -64,12 +66,44 @@ void AchievementManager::Init()
     m_queue.Reset("AchievementManagerQueue", [](const std::function<void()>& func) { func(); });
     m_image_queue.Reset("AchievementManagerImageQueue",
                         [](const std::function<void()>& func) { func(); });
+    LoadApprovedList();
     if (HasAPIToken())
       Login("");
     INFO_LOG_FMT(ACHIEVEMENTS, "Achievement Manager Initialized");
   }
 }
 
+void AchievementManager::LoadApprovedList()
+{
+  m_queue.EmplaceItem([this]() {
```

Comment:
This is a race condition on `m_ini_root`, you cannot guarantee that the load is done by the time `FilterApprovedPatches()` is called. Personally I would just move the load of the json into the constructor to work around that...

Reply:
Not sure I understand how putting it in the constructor would fix that, but okay.

Reply:
To be clear: Don't put it in the queue, just load it synchronously.